engine/client/drivers/docker.go

```diff
@@ -185,19 +205,45 @@ func resolveImageID(imageRef string) (string, error) {
 }
 
 func garbageCollectEngines(ctx context.Context, log *slog.Logger, engines []string) {
+	shouldCleanup := shouldCleanupEngines()
+	val := os.Getenv("DAGGER_LEAVE_OLD_ENGINE")
+
+	// Enhanced logging for debugging
+	log.Info("evaluating engine cleanup",
+		"raw_env_value", val,
+		"shouldCleanup", shouldCleanup,
+		"engineCount", len(engines))
+
+	// Log each engine being considered
+	for i, engine := range engines {
+		log.Info("found engine",
+			"index", i,
+			"name", engine)
+	}
+
+	if !shouldCleanup {
+		log.Info("preserving engines due to DAGGER_LEAVE_OLD_ENGINE",
+			"value", val,
+			"engineCount", len(engines))
+		return
+	}
```

Comment:
this block doesn't make a lot of sense since this function is **only** called when there's something to cleanup,  so this path will currently never be executed.